### PR TITLE
AP_Proximity: Add backend for scripted Lua Proximity Driver 

### PR DESCRIPTION
**Pull request by rishabsingh3003**
This PR adds Lua backend so that we can create a scripted proximity driver
I have also added an example of a new Benewake sensor: RDS02C

Thanks to @yuri-rage who actually did most of the work on the Proximity side of things! I just fixed a few things from his branch.

### CODE REVIEW

**Comment by yuri-rage** [comment]
Great to see this re-invigorated! Glad to contribute a bit, even if my initial commits were slightly rough!

File: libraries/AP_Proximity/AP_Proximity_Backend.h [hunk 1]
```diff
@@ -42,6 +42,16 @@ class AP_Proximity_Backend
     // handle mavlink messages
     virtual void handle_msg(const mavlink_message_t &msg) {}
 
+#if AP_SCRIPTING_ENABLED
+    // handle Lua obstacle messages
+    virtual bool set_distance_min_max(float min, float max) { return false; }
+    virtual bool handle_script_distance_msg(float dist_m, float azimuth_deg, float elevation_deg, bool push_to_boundary) { return false; }
```
Comment (rmackay9) [inline libraries/AP_Proximity/AP_Proximity_Backend.h:48]: I think we should probably rename "azimuth" to "yaw" or "heading".  "azimuth" is a fine word on it's own but we should try and avoid synonyms where possible.

File: libraries/AP_Proximity/AP_Proximity_Params.cpp [hunk 2]
```diff
@@ -8,7 +8,7 @@ const AP_Param::GroupInfo AP_Proximity_Params::var_info[] = {
     // @Param: _TYPE
     // @DisplayName: Proximity type
     // @Description: What type of proximity sensor is connected
-    // @Values: 0:None,7:LightwareSF40c,2:MAVLink,3:TeraRangerTower,4:RangeFinder,5:RPLidarA2,6:TeraRangerTowerEvo,8:LightwareSF45B,10:SITL,12:AirSimSITL,13:CygbotD1
+    // @Values: 0:None,7:LightwareSF40c,2:MAVLink,3:TeraRangerTower,4:RangeFinder,5:RPLidarA2,6:TeraRangerTowerEvo,8:LightwareSF45B,10:SITL,12:AirSimSITL,13:CygbotD1, 14:LuaScripting
```
Comment (rmackay9) [inline libraries/AP_Proximity/AP_Proximity_Params.cpp:11]: Another small naming thing, looking at other drivers we normally call it "Scripting" (e.g. skip the "Lua" part)

File: libraries/AP_Proximity/AP_Proximity_Lua.cpp [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+    // extract yaw and pitch from Obstacle Vector
+    const float yaw = wrap_360(degrees(atan2f(obstacle.y, obstacle.x)));
+    const float pitch = wrap_180(degrees(M_PI_2 - atan2f(obstacle.xy().length(), obstacle.z)));
+
+    // now simply handle as a distance msg
+    return handle_script_distance_msg(obstacle.length(), yaw, pitch, push_to_boundary);
+}
+
+#endif // HAL_PROXIMITY_ENABLED && AP_SCRIPTING_ENABLED
```
Comment (rmackay9) [inline libraries/AP_Proximity/AP_Proximity_Lua.cpp:116]: nitpick: might be nice to put an newline at the end of the file.. some editors will add them constantly.

File: libraries/AP_Scripting/examples/proximity_can_driver_RDS02C.lua [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+local sensor_min_range = 0    --minimum range of sensors in meters
+local update_rate_ms    = 25    -- update rate (in ms) of the driver
+local update_rate_error_ms  = 5000 -- update rate incase there is a fatal error (in ms)
+local debug_enable = false    -- true to enable debug messages
+
+-- Global variables (DO NOT CHANGE)
+local param_num_lua_backend = 14         -- parameter number for lua rangefinder
+local lua_backend                        -- store lua prx backend here
+local lua_driver_found = false           --true if user has configured lua prx backend
+local can_driver = CAN:get_device(5)
```
Comment (rmackay9) [inline libraries/AP_Scripting/examples/proximity_can_driver_RDS02C.lua:15]: maybe add a comment to explain the "5"?

**Pull request closed without merging**